packages/react-native-reanimated/apple/reanimated/apple/keyboardObserver/REAKeyboardEventObserver.mm

```diff
@@ -202,37 +186,46 @@ - (void)keyboardWillChangeFrame:(NSNotification *)notification
   CGRect beginFrame = [[userInfo objectForKey:UIKeyboardFrameBeginUserInfoKey] CGRectValue];
   CGRect endFrame = [[userInfo objectForKey:UIKeyboardFrameEndUserInfoKey] CGRectValue];
   NSTimeInterval animationDuration = [[userInfo objectForKey:UIKeyboardAnimationDurationUserInfoKey] doubleValue];
-  CGSize windowSize = [[[UIApplication sharedApplication] delegate] window].frame.size;
+  auto window = [[[UIApplication sharedApplication] delegate] window];
+  CGSize windowSize = window.frame.size;
+
+  beginFrame = [window convertRect:beginFrame fromCoordinateSpace:window.screen.coordinateSpace];
+  endFrame = [window convertRect:endFrame fromCoordinateSpace:window.screen.coordinateSpace];
 
-  _initialKeyboardHeight = windowSize.height - beginFrame.origin.y;
+  _initialKeyboardHeight = CGRectGetMaxY(beginFrame) > 0 ? windowSize.height - beginFrame.origin.y : 0;
   _targetKeyboardHeight = windowSize.height - endFrame.origin.y;
 
+  bool prefersCrossFade = false;
+#if __has_include(<UIKit/UIAccessibility.h>)
+  prefersCrossFade = UIAccessibilityPrefersCrossFadeTransitions();
+#endif
+  if (prefersCrossFade && beginFrame.origin.y == 0) {
+    _initialKeyboardHeight = 0;
+  }
```

Comment:
I'd definitely add a comment with explanation here as well. From what I've seen and tinkered with, only the begin frame seems to be off, the `endFrame` (and thus `targetKeyboardHeight`) is fine.

Reply:
That is what I have seen as well. And in fact, this code has no effect anymore after I added the guard above to check `CGRectGetMaxY` for other reasons. It may still make sense to have this here to explicitly set the height to `0`, but if you think it should just be removed that is fine as well.

Reply:
Oh yeah I forgot - the CGRectGetMaxY would also need some explanation haha
I couldn't quite wrap my head around what it does here